### ISSUE TITLE
clarify what is a valid topic

### ISSUE THREAD

**Issue by tomkralidis** [wmo-im/wis2-topic-hierarchy#137]
As discussed with @golfvert, we need to clarify whether "partial" topics are deemed valid and can/should be used or not.

For example, `origin/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations/synop` is a valid topic.

Should `origin/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations` be considered a valid topic as well?  This means a topic without a leaf?

We would need to update the specification to be clear in this regard (and whether Requirement 1B needs to be updated/augmented.  In addition, we would need to update the artefacts made available on schemas.wmo.int for the WTH CV bundle/lookup.

**Comment by golfvert** [comment]
For me, only "leaf" topics are usable. Otherwise `origin/a/wis2/ca-eccc-msc/data/core/weather` would be also usable...
If we have agreed sublevels in eg `origin/a/wis2/ca-eccc-msc/data/core/weather/surface-based-observations` then, one of them MUST be used.

**Comment by antje-s** [comment]
I agree, if there are subtopics, one of the subtopics should be used, otherwise we will have many messages published to the top topics. If there are no suitable subtopic values (e.g. for "new" data), experimental should be used and the required new subtopic values should be added as soon as possible to main WTH. As soon as the new values are available, the publish must be switched back to the main branch of the WTH

**Comment by josusky** [comment]
Publishing only to "leafs" sounds good to me.

**Comment by tomkralidis** [comment]
Adding @sebvi for feedback / validation.

**Comment by david-i-berry** [comment]
What happens if there is a new CAP alert where none of the leaves are appropriate? If we do not allow publication on a non-leaf node we need to have 100 % coverage from day 1.

**Comment by amilan17** [comment]
If publication down to the last leaf is required, what should be done in the following scenario? 
1. data is already being published to topic `.../discipline/topic-x/topic-a` 
2. New sub-topics of topic-a are proposed `.../discipline/topic-x/topic-a/1` and `...topic-x/topic-a/2`

**Comment by amilan17** [comment]
> > Can you tell us which one ? Looking at what ECMWF is publishing I see only "leaf"
> 
> I don't see any examples on the explorer, but I think it would be good to double check with TT-NWPMD.

I spoke with Yuki and this will be on the agenda of the TT-NWPMD's next meeting.

**Comment by josusky** [comment]
I am still OK with publishing to leaf topics only. 

Splitting of a leaf to multiple leaves is a change management issue that is largely unrelated.
We can propose some procedures that would make such change less of a "breaking change" than it may seem, such as:
* recommend subscribers / data consumers to use wildcard (`#`) at the end of the subscribed topics
* require the proposer of such a split to make a throw analysis to ensure that (1) there will be a matching sub-topic for all data currently published under the current leaf or there is no data at all, and (2) that the subscribers, if any, use the aforementioned wildcard - a GB could help with the latter, I guess

**Comment by sebvi** [comment]
Sorry for not responding back earlier. I don't see a use case where you would not publish up to the leaf. 
We discussed this in our TT-NWPMD meeting and the reason we stopped at the hierarchy level 12 is because going further would become too specific and restrictive and could force institutions to publish at a branch level instead of the leaf.

I support publishing to leaf only

**Comment by tomkralidis** [comment]
TT-WISMD 2024-06-28:
- how to manage the change
  - extending / adding leaves
  - adding a new subtopic invalidates the parent topic (breaking change)
  - for ESDs with no subtopics yet, can they use experimental?
    - may be risky (could turn experimental into a free-for-all)
  - provide a transition period between two FTs
 - perhaps too early to address in the lifespan of WIS2

ACTION: @tomkralidis to raise at WIS2 Architecture.

**Comment by golfvert** [comment]
Did we achieve a conclusion on this ?

**Comment by tomkralidis** [comment]
We will revisit / discuss at the next [TT-WISMD meeting](https://github.com/wmo-im/tt-wismd/wiki/Meeting-2024-09-17).

**Comment by tomkralidis** [comment]
Noting TT-WISMD 2024-06-28, after discussion with @golfvert, and given adding subtopics and the resulting invalidation / breaking changes (i.e. beyond an amendment), we agree that a suitable alternative may be:

- publishers SHOULD use leaf topics
- publishers MAY use non-leaf topics

I've updated the PR in #137 for good measure, but we first need @wmo-im/tt-wismd consensus in this regard.

**Comment by tomkralidis** [comment]
TT-WISMD 2025-02-18:
- @wmo-im/tt-wismd to review/assess

**Comment by josusky** [comment]
In the WTH we neither mention **leaf** nor **tree**. The latter is shocking to me. This is mainly a note for @amilan17 that we might need to introduce this terminology (metaphors) into the document.

**Comment by golfvert** [comment]
@josusky it is just a quick way to illustrate the two options. We don't plan to use this officially in the manual.
